### DISCUSSION TITLE
use stb_image.h's stbi_load func to read Image, the data format is HWC or CHW?

### DISCUSSION THREAD

**Discussion by zjykzj** [nothings/stb#1285] [Q&A] [answered]
**Describe the bug**

Hi @nothings , first thank you very much for our easy-to-use implementation.

I have a question about data format which was read using stb_image.h's stbi_load func.

It's data format is HWC or CHW? For example, OpenCV C++ read image using HWC data format. I didn't find any relevant information in the document, thank you very much if you can reply to me directly

**Comment by nothings** [comment, marked as answer]
Line 153 of stb_image.h:

>// The pixel data consists of *y scanlines of *x pixels,
>// with each pixel consisting of N interleaved 8-bit components; the first
>// pixel pointed to is top-left-most in the image. There is no padding between
>// image scanlines or between pixels, regardless of format.

**Reply by zjykzj** [reply]
> Line 153 of stb_image.h:
> 
> > // The pixel data consists of *y scanlines of *x pixels,
> > // with each pixel consisting of N interleaved 8-bit components; the first
> > // pixel pointed to is top-left-most in the image. There is no padding between
> > // image scanlines or between pixels, regardless of format.

Good, Thank you for your reply ~~~ 

That means the data format is HWC (for example RGBRGBRGBRGB...) like OpenCV does

**Reply by nothings** [reply]
correct
